### PR TITLE
Scrum 17

### PR DESCRIPTION
**Pull request by lebenpdx**
Reduced the amount of players being initially rendered from all players in the league to 20 random players. Also created a drop down of all of the teams in the league; selecting one will display all players on that team.

Implemented parallel api requests to speed up loading times. Also implemented caching to reduce load times in between switching back and forth between the same team multiple times.

chrome_agpDibL264

### CODE REVIEW

File: cs464-group-project/src/pages/Players/PlayersMainPage.js [hunk 1]
```diff
@@ -62,52 +66,102 @@ export function Player() {
...
+        setPlayers(randomPlayers);
+      } else {
+        if (teamcache[team]) {
+          setPlayers(teamcache[team]);
+        } else {
+          const response = await getAllPlayersByTeam(team); // Fetch players for the selected team
+
+          if (!response.ok) {
+            setPlayers(response.player || []);
+            teamcache = response.player;
```
Comment (txiong002) [inline cs464-group-project/src/pages/Players/PlayersMainPage.js]: getting an warning saying 'teamchace' is constant.  If we're changing the variable value, would a 'let teamcache' work instead?

**Review by txiong002 — CHANGES_REQUESTED** [review]
Besides the warning, everything else looks good.  The page definitely loads a lot faster now.

File: cs464-group-project/src/pages/Players/PlayersMainPage.js (hunk 1 above)
Comment (lebenpdx) [reply cs464-group-project/src/pages/Players/PlayersMainPage.js]: Should be fixed now

**Review by txiong002 — APPROVED** [review]
Approved

**Pull request merged**